### ISSUE TITLE
[Enhancement] PinsSource on Forms.Map

### ISSUE THREAD

**Issue by hartez** [xamarin/Xamarin.Forms#1708]
## Rationale

Forms.Maps does not currently have a way to bind a list of Pins on a Map.

## Implementation

Add a bindable property PinsSource to Map:

```
public static readonly BindableProperty PinsSourceProperty =
	BindableProperty.Create(nameof(PinsSource), typeof(IList), typeof(Map), default(IList));

public IList PinsSource
{
	get { return (IList)GetValue(PinsSourceProperty); }
	set { SetValue(PinsSourceProperty, value); }
}	
```

And a PinTemplate property:

```
public static readonly BindableProperty PinTemplateProperty = BindableProperty.Create("PinTemplate", typeof(DataTemplate), typeof(Map);

public DataTemplate PinTemplate
{
	get { return (DataTemplate)GetValue(PinTemplateProperty); }
	set { SetValue(PinTemplateProperty, value); }
}
```

The PinTemplate is responsible for converting the objects in the PinsSource into `Xamarin.Forms.Maps.Pin` objects.

Setting the `PinsSource` will clear out any existing pins and update the pin collection to match the source. 
	
## Backward Compatibility

This is a new property, so there should not be any backward compatibility issues.

**Comment by jassmith** [comment]
Should we maybe use a DataTemplate + IEnumable<object> instead to avoid the inevitable issue of people then being forced to use a Forms type in their ViewModel to make this work?

Also the behavior of what this does to the Pins collection needs to be well defined.

**Comment by davidortinau** [comment]
@jsuarezruiz this is for you

**Comment by jassmith** [comment]
Hey @jsuarezruiz, you still working on this? If not we can move it back to the ready for work column

**Comment by jsuarezruiz** [comment]
@jassmith  I focused earlier on the CornerRadius of the BoxView and I'm still working on it. At the moment I have a version for Android, iOS and UWP. I need to implement in GTK, macOS and WPF. On the other hand, in the current implementation I use ViewCell in the DataTemplate and I'm not sure if we wants to avoid or there's no problem.
I try to have it as soon as possible, although if someone from the Xamarin.Forms Team or the community wants to continue/review it, I share what I have and I take another one.

**Comment by jassmith** [comment]
No rush, Im just doing check ins to make sure we don't lose track of things or get people duplicating efforts, if you are still working on it super :)

If you want to post up your partial work so far, great :) We would take the implementation on Android/iOS/UWP as is.

**Comment by PureWeen** [comment]
> If you want to post up your partial work so far, great :) We would take the implementation on Android/iOS/UWP without waiting on the others.

@jsuarezruiz 

thoughts?

**Comment by samhouts** [comment]
@jsuarezruiz Any progress? Thanks!

**Comment by jsuarezruiz** [comment]
Oops, sorry for taking time to respond. What I was doing is available at: [https://github.com/XamarinFormsCommunity/Xamarin.Forms/tree/feature/1708-pinssource](https://github.com/XamarinFormsCommunity/Xamarin.Forms/tree/feature/1708-pinssource) 
I will review it and continue.

**Comment by samhouts** [comment]
@jsuarezruiz Monthly ping! How is it going? Thanks!

**Comment by hugodantas** [comment]
Hi! I suppose we have this feature rolling out soon, right?
How is it going? Thank you so much!

**Comment by andreinitescu** [comment]
I don't get it, what is the actual specification for this implementation? How does the Xamarin Forms team wants this implemented? I put quite an effort into #4269 but apparently it's not how it should be.
